### DISCUSSION TITLE
Time entity node (beta)

### DISCUSSION THREAD

**Discussion by zachowj** [zachowj/node-red-contrib-home-assistant-websocket#988] [General]
placeholder

**Comment by Axaroz** [comment]
Hey! I noticed that if you restart node-red then it doesnt send a new msg when in "listen for change mode". Would be wonderful with a send on connect option for it similar to how some of the other nodes like poll state have for example. 

Thanks for a great addition.

**Reply by zachowj** [reply]
A current workaround would be to use an inject node with "Inject once after 0.1 seconds, then" into a time entity node with its mode set to "get value"

**Reply by schford** [reply]
Could you possibly elaborate please as to how you would set up the inject node? - am fairly new to this all. Thanks

**Reply by zachowj** [reply]
![image](https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/37859597/6ab5036d-f93b-4eed-a936-36e9ad93b961)

```
[{"id":"4f9cfa186df651b4","type":"ha-time-entity","z":"120358abd7c22d30","name":"","version":0,"debugenabled":false,"inputs":1,"outputs":1,"entityConfig":"","mode":"get","value":"payload","valueType":"msg","outputProperties":[{"property":"payload","propertyType":"msg","value":"","valueType":"value"}],"x":636,"y":7728,"wires":[[]]},{"id":"dcce141a2158161e","type":"inject","z":"120358abd7c22d30","name":"","props":[],"repeat":"","crontab":"","once":true,"onceDelay":0.1,"topic":"","x":498,"y":7728,"wires":[["4f9cfa186df651b4"]]}]
```

**Comment by lougreenwood** [comment]
I get the following error when trying to set a value for `time`: `"TypeError: text.split is not a function or its return value is not iterable"`.

![Screenshot 2024-03-08 at 13 31 14](https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/12051736/b93f9117-5ff6-4066-a5b3-ee019b1d6451)
![Screenshot 2024-03-08 at 13 31 01](https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/12051736/d9289d1b-34d4-4a74-964b-ba3fd178b3a7)
